src/modules/assessments/components/AssessmentForm.tsx

```diff
@@ -117,11 +114,24 @@ const AssessmentForm: React.FC<Props> = ({
   useEffect(() => {
     if (!canEdit) return;
     if (assessment && !assessment.inProgress) setLocked(true);
-  }, [assessment, canEdit]);
+  }, [assessment, canEdit]); // re-runs after assessment is refetched, which leads to assessment re-locking after submit
+
+  // Choose the FormDefiniton to use for rendering, and filter it down based on client attributes (Data Collected About rules).
+  const definition = useMemo(() => {
+    let fd = definitionProp;
+    if (!locked && assessment && !assessment.inProgress) {
+      // When editing a non-WIP assessment, we want to use the most recent version of the Form Definition.
+      // If the most recent version differs from the version that was most recently used to updated it, then 'upgradedDefinitionForEditing' is present.
```

Comment:
I remember some back-and-forth in design sessions about whether it would be confusing to users that in-progress forms are still opened using the old form version. We discussed adding a button to prompt the user to upgrade to the latest version, and decided against it since it seems like too much effort for an edge case.

I wonder, though, if just adding a brief sentence to the top of the form, something like, "A new version of this assessment has been published since the draft was started. To upgrade, delete this draft and begin again" could be a low effort way to help clarify things.

Or we can just push what you have and see if users actually get confused 🤷‍♀️

Reply:
We landed on not showing anything, I think because of the underlying goal to make Provider-facing interfaces as simple as possible. If 90% of the time if the change is to the wording of a question, it doesn't really matter––they might be alarmed or confused by seeing a new message, when it is (most likely?) not relevant (or actionable) to them as a Provider.